lib/http_router/generator.rb

```diff
@@ -36,13 +36,13 @@ def initialize(route, path, validation_regex = nil)
             instance_eval <<-EOT, __FILE__, __LINE__ + 1
             def generate(args, options)
               generated_path = \"#{code}\"
-              #{validation_regex.inspect}.match(generated_path) ? URI.escape(generated_path) : nil
+              #{validation_regex.inspect}.match(generated_path) ? URI::DEFAULT_PARSER.escape(generated_path) : nil
```

Comment:
Looking at [the 2.7 docs and source code](https://ruby-doc.org/stdlib-2.7.0/libdoc/uri/rdoc/URI/Escape.html#method-i-escape), it appears that `URI::DEFAULT_PARSER.escape` is what the deprecated `URI.escape` method is doing. The docs recommend using a different method, should we follow that recommendation instead? I believe `CGI.escape` will do what we want.

Reply:
So my initial thought was to use either `CGI` or `ERB::Util`,  however, as you can see from the following results, they are not the dame as URI:

```irb
URI.escape('https://www.squaremouth.com/travel-insurance-providers/global-underwriters/reviews')
=> "https://www.squaremouth.com/travel-insurance-providers/global-underwriters/reviews"

CGI.escape('https://www.squaremouth.com/travel-insurance-providers/global-underwriters/reviews')
=> "https%3A%2F%2Fwww.squaremouth.com%2Ftravel-insurance-providers%2Fglobal-underwriters%2Freviews"

ERB::Util.url_encode('https://www.squaremouth.com/travel-insurance-providers/global-underwriters/reviews')
=> "https%3A%2F%2Fwww.squaremouth.com%2Ftravel-insurance-providers%2Fglobal-underwriters%2Freviews"
```

I ended up going with `URI::DEFAULT_PARSER.escape` based on [this comment](https://github.com/joshbuddy/http_router/pull/47#issuecomment-262235430).

I also found that CGI was not a complete replacement for URI based on this conversation [here](https://stackoverflow.com/questions/2824126/whats-the-difference-between-uri-escape-and-cgi-escape/13059657#13059657).